src/index.js

```diff
@@ -133,3 +126,171 @@ export class WealthscopeSdk {
+   * @param {string} url 
+   * @param {object} body 
+   * @returns {Promise}
+   */
+  put(url, body) {
+    const postOptions = this._getFetchOptions('PUT', body);
+    return fetch(this._constructUrl(url), postOptions)
+      .then(response => response.json())
+      .catch(err => console.log(err));
+  }
+
+  /**
+   * Fetch call returning a Promise, requires a URL rout, a body and a valid
+   * token.
+   * NOTE: Body needs to be a JSON object here, it gets stringified when it is
+   * added to the options object for the fetch.
+   * @param {string} url 
+   * @param {object} body 
+   * @returns {Promise}
+   */
+  post(url, body) {
+    const postOptions = this._getFetchOptions('POST', body);
+    return fetch(this._constructUrl(url), postOptions)
+      .then(response => response.json())
+      .catch(err => console.log(err));
+  }
+
+  /**
+   * Fetch call returning a Promise, requires a URL rout, a body and a valid
+   * token.
+   * NOTE: Body needs to be a JSON object here, it gets stringified when it is
+   * added to the options object for the fetch.
+   * @param {string} url 
+   * @param {object} body 
+   * @returns {Promise}
+   */
+  del(url, body) {
+    const postOptions = this._getFetchOptions('DELETE', body);
+    return fetch(this._constructUrl(url), postOptions)
+      .then(response => response.json())
+      .catch(err => console.log(err));
+  }
+
+  /**
+   * Generates a URL, and ensures a trailing '/'
+   * @param {string} url
+   * @returns {string} 
+   * @private
+   */
+  _constructUrl(url) {
+    // trailing slash is important in order for `fetch` to work
+    return urlJoin(this.opts.wealthscopeUrl, url) + '/'
+  }
+
+  /**
+   * This function constructs the options object that will be used by the
+   * various fetch calls below.
+   * NOTE: method must be in ALL CAPS, body must be a json object
+   * @param {string} method 
+   * @param {object} body 
+   * @returns {object}
+   * @private
+   */
+  _getFetchOptions(method, body) {
+    // An options object with empty headers
+    const fetchOption = {
+      headers: {}
+    };
+
+    // The Method portion of the object
+    fetchOption.method = method;
+
+    // A check to make sure token is present before adding it to the options
+    // return an error if token isn't present
+    if (this.token != null) {
+      fetchOption.headers.Authorization = 'JWT ' + this.token;
+    } else {
+      throw new Error('You must be logged in to perform this action.');
+    }
+
+    // A check for a body before it's added to the options, allows the same
+    // construction functon to be used for calls without a body component
+    // Returns the object without the body if body is empty, adds the body to
+    // the object if it's present
+    if (body != null) {
+      fetchOption.headers['Content-Type'] = 'application/json';
+      fetchOption.body = JSON.stringify(body);
+    }
+
+    return fetchOption;
+  }
+
+}
+
+window.API = WealthscopeApiClient;
```

Comment:
remove this line, it was only intended for debugging. it shouldn't make it to production code.